Automattic/wp-calypso
# Remove before offer reset components that are not used anymore

### Changes proposed in this Pull Request

This PR removes code that was used before the offer reset and became deprecated.

Follow up of #47466.

### Testing instructions

The name of the game is to make sure that what's been deleted is not used elsewhere in the code. Download the PR and in your editor, check that you don't find mentions of:

- `plans-skip-button`
- `plansLanding`
- `plansSelection`
- `completeFlow`

And that the following files are not imported anywhere:

- `client/jetpack-connect/plans-grid.jsx`
- `client/jetpack-connect/plans-landing.jsx`
- `client/jetpack-connect/plans-placeholder.js`
- `client/jetpack-connect/plans.jsx`
- `client/state/jetpack-connect/actions/complete-flow.js`

Also, visit the _Plans_ page and verify that it looks as expected, without any error in the console.
